data/core/view.lua

```diff
@@ -46,51 +47,77 @@ function View:get_name()
   return "---"
 end
 
-
+-- Returns `y, x` for compatibility
 function View:get_scrollable_size()
-  return math.huge
+  return math.huge, 0 -- TODO: invert y and x
 end
 
 
-function View:get_scrollbar_rect()
-  local sz = self:get_scrollable_size()
-  if sz <= self.size.y or sz == math.huge then
-    return 0, 0, 0, 0
+function View:get_scrollbar_rect() -- TODO: make plugins use `View:get_scrollbars_rect`
+  local rect = self:get_get_scrollbars_rect()
+  return rect.x, rect.y, rect.w, rect.h
+end
+
+function View:get_scrollbars_rect()
+  local v_sizes = { x = 0, y = 0, w = 0, h = 0 }
+  local h_sizes = { x = 0, y = 0, w = 0, h = 0 }
+  local v_sz, h_sz = self:get_scrollable_size()
+  h_sz = h_sz or 0 -- FIXME: not every subclass returns the second value
+
+  if v_sz > self.size.y and v_sz ~= math.huge then
+    local h = math.max(20, self.size.y * self.size.y / v_sz)
+    v_sizes.x = self.position.x + self.size.x - style.scrollbar_size
+    v_sizes.y = self.position.y + self.scroll.y * (self.size.y - h) / (v_sz - self.size.y)
+    v_sizes.w = style.scrollbar_size
+    v_sizes.h = h
   end
-  local h = math.max(20, self.size.y * self.size.y / sz)
-  return
-    self.position.x + self.size.x - style.scrollbar_size,
-    self.position.y + self.scroll.y * (self.size.y - h) / (sz - self.size.y),
-    style.scrollbar_size,
-    h
+
+  if h_sz > self.size.x and h_sz ~= math.huge then
+    local w = math.max(20, self.size.x * self.size.x / h_sz)
+    h_sizes.x = self.position.x + self.scroll.x * (self.size.x - w) / (h_sz - self.size.x)
+    h_sizes.y = self.position.y + self.size.y - style.scrollbar_size
+    h_sizes.w = w
+    h_sizes.h = style.scrollbar_size
+  end
+
+  return v_sizes, h_sizes
 end
 
 
 function View:scrollbar_overlaps_point(x, y)
-  local sx, sy, sw, sh = self:get_scrollbar_rect()
-  return x >= sx - sw * 3 and x < sx + sw and y >= sy and y < sy + sh
+  local v, h = self:get_scrollbars_rect()
+  local v_overlap = x >= v.x - v.w * 3 and x < v.x + v.w and y >= v.y and y < v.y + v.h
+  local h_overlap = x >= h.x and x < h.x + h.w and y >= h.y - h.h * 3 and y <= h.y + h.h
+  return v_overlap, not v_overlap and h_overlap -- precedence to vertical scrollbar
 end
 
 
 function View:on_mouse_pressed(button, x, y, clicks)
-  if self:scrollbar_overlaps_point(x, y) then
-    self.dragging_scrollbar = true
-    return true
-  end
+  self.dragging_v_scrollbar, self.dragging_h_scrollbar = self:scrollbar_overlaps_point(x, y)
+  self.dragging_scrollbar = self.dragging_v_scrollbar -- TODO: make plugins use `self.dragging_v_scrollbar`
```

Comment:
I advice to not touch the plugins and keep `self.dragging_scrollbar` just for the vertical scrolling. The rule will be that scrolling without "h" in their name will be about vertical scrolling.

I suggest that in the code we drop `self.dragging_v_scrollbar` in favor of `self.dragging_scrollbar`. I recognize we violate the symmetry between h and v but v came first, historically.